### PR TITLE
Framework: update to react 16 (take three)

### PR DESCRIPTION
**Pull request by samouri**
Everyone knows the third time is the charm.

to do:
1. [x] interpolate components: https://github.com/Automattic/interpolate-components/pull/12, https://github.com/Automattic/wp-calypso/pull/19658
2. [x] fix ssr: https://github.com/Automattic/wp-calypso/pull/20267

To Test and Verify
------
1. local dev
2. docker. note: ssr will be broken unless `secrets.json` is configured. 
3. there should be no reference to React < 16 within our shrinkwrap

### CODE REVIEW

**Comment by kwight** [comment]
Note that `notifications-panel` (the `react16` branch aligned with this update) is now at `2.1.1`.

**Comment by sirreal** [comment]
> Note that notifications-panel (the react16 branch aligned with this update) is now at 2.1.1.

@kwight `2.1.1` does not seem to be published yet 😬

**Comment by sirreal** [comment]
This had become quite out of date.

Rebased, `notifications-panel@2.1.0`, `update-deps`, force push.

**Comment by sirreal** [comment]
Rebased again, `notifications@2.1.0` was causing errors. Back to `2.0.0`

**Comment by kwight** [comment]
Ack, will verify 2.1.1.

**Comment by kwight** [comment]
`notifications-panel@2.1.1` published 👍

**Comment by samouri** [comment]
I'm basing this off the ssr fix now

**Comment by samouri** [comment]
🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 
⭐️ ⭐️ ⭐️ ⭐️ ⭐️ ⭐️   ready for testing  ⭐️ ⭐️ ⭐️ ⭐️ ⭐️ ⭐️ ⭐️ ⭐️ ⭐️
🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑 🐑

**Comment by sirreal** [comment]
Running tests now produces far fewer warnings, which is nice. Now we have a new annoyance:

> console.error node_modules/fbjs/lib/warning.js:33
> Warning: React depends on requestAnimationFrame. Make sure that you load a polyfill in older browsers. http://fb.me/react-polyfills

Relevant conversation seems to be happening here: https://github.com/facebook/jest/issues/4545. Just a warning, so nothing blocking and we can ship a fix later.

File: package.json [hunk 1]
```diff
@@ -106,7 +106,7 @@
     "ms": "0.7.1",
     "name-all-modules-plugin": "1.0.1",
     "node-sass": "4.5.3",
-    "notifications-panel": "1.2.7",
+    "notifications-panel": "2.0.0",
```
[the remark this answers is not in the document]
Comment (samouri) [reply package.json]: also React 16.2!

[the remark this answers is not in the document]
Comment (sirreal) [reply package.json]: Love those `<>Fragments</>` 🎉

File: package.json [hunk 2]
```diff
@@ -277,7 +277,7 @@
     "prettier": "github:automattic/calypso-prettier#717e62ee",
     "pretty-bytes": "4.0.2",
     "react-codemod": "reactjs/react-codemod",
-    "react-test-renderer": "15.6.2",
+    "react-test-renderer": "16.0.0",
```
Comment (ockham) [inline package.json]: Also bump to 16.2.0?

**Comment by samouri** [comment]
I've had to bump back down to 2.0 due to issues with`esnext`

**Comment by ockham** [comment]
> I've had to bump back down to 2.0 due to issues with`esnext` 

My bad. Context: https://github.com/Automattic/wp-calypso/pull/19698

To fix, we can:
* either merge #19698 
* or change back `esnext` to `module` in https://github.com/Automattic/notifications-panel
* or add a build step, publish that to npm, and set  `main` to point to the dir that holds the transpiled code

**Comment by samouri** [comment]
> To fix, we can:

Am I right in thinking that can all happen after this PR?

**Comment by samouri** [comment]
after upgrading a few packages (and then redowngrading notif-panel), i'm doing another smoke test and then asking for an E2E test from flow-patrol

**Comment by ockham** [comment]
> Am I right in thinking that can all happen after this PR?

Oh, absolutely! Sry for not having been clear 😄

**Comment by ockham** [comment]
>> Am I right in thinking that can all happen after this PR?

> Oh, absolutely! Sry for not having been clear 😄 

As pretty much anything that doesn't make Calypso go up in 🔥s at this point 😄

**Comment by bsessions85** [comment]
E2E tests passed

**Pull request merged**